src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -365,8 +369,23 @@ private void OpenIndexes()
 
                     exceptionAggregator.Execute(() =>
                     {
-                        var index = Index.Open(indexId, _documentDatabase);
-                        _indexes.Add(index);
+                        try
+                        {
+                            var index = Index.Open(indexId, _documentDatabase);
+                            _indexes.Add(index);
+                        }
+                        catch (Exception e)
+                        {
+                            // TODO arek: I think we can ignore auto indexes here, however for static ones try to retrieve names
+                            var fakeIndex = new FaultyInMemoryIndex(indexId);
```

Comment:
Yes, we need it.
Note that a failure of a single index (or all indexes) is fine, and will result in the faulty index showing up there.
We try to load each index, and we won't fail the whole open process if we can't open an index.
https://github.com/ayende/ravendb/blob/v4.0/src/Raven.Server/Documents/Indexes/IndexStore.cs#L378

Only if the _entire indexing process is broken_ (the indexing directory is not found, for example, or inaccessible), or something like this. Will we abort it.